### DISCUSSION TITLE
There is already an object named 'Bookmarks' in the database

### DISCUSSION THREAD

**Discussion by Aoshua** [elsa-workflows/elsa-core#1023] [Q&A] [answered]
I have two projects using Elsa at the moment. One that is both the Designer and the Server, to build & test workflows. Then another which is a Web API where ultimately all the workflows will be run in production.

After creating workflows with the designer, trying to load the Web API project results in the following exception: `Microsoft.Data.SqlClient.SqlException: 'There is already an object named 'Bookmarks' in the database`
(I didn't used to get this issue with Elsa 1.5.x)

To access the workflow definitions and such in my Web API project, I have the following line in the `Startup.cs` :
`.UseEntityFrameworkPersistence(ef => ef.UseSqlServer(Configuration.GetConnectionString("EXSContext")))`
Is there an option here to specify that the tables have already been created, so there is no need to run a migration? Or is there something else I can do to allow it to work with the existing tables rather than complaining that they are already there?

**Comment by BrianHumlicek** [comment]
I had the same thing happen with the current build.  I 'fixed' it by deleting the db file.  If you have existing data you want to save, this isn't much help.

**Comment by sfmskywalker** [comment, marked as answer]
EF Core Migrations should be able to see already that the target DB has the up-to-date schema. But what might have happened is that your DB contains an older migration. This happens once in a while when I **regenerate migrations** while Elsa 2.0 is not yet officially released (once released, I will maintain the released migrations and generate Update steps).

So in your case your problem may be fixed by deleting the DB as is also suggested by Brian.

On a different note: is there a reason the first project needs to host both the Dashboard and the Server? Why not let that project just be the dashboard that points to the server (your second project)?

You could then even consider just running the Dashboard from a Docker container and not even bother maintaining a separate project (unless you're doing some customizations).

E.g. the following docker command starts the dashboard on port 9000 while allowing you to have it point to your Elsa Server project listening on port 15000:

```bash
docker run -t -i -e ELSA__SERVER__BASEURL='http://localhost:15000' -p 9000:80 elsaworkflows/elsa-dashboard:latest
```

Anyway, just a tip. There shouldn't be any issues having two servers pointing to the same database (except for what is reported here: #1020).

**Reply by Aoshua** [reply]
Simply running the designer in a Docker container while my API works as the sever is amazing 🤯 This should fix the issue I was seeing of two separate applications trying to access the same tables.
